sdk/textanalytics/Azure.AI.TextAnalytics/api/Azure.AI.TextAnalytics.netstandard2.0.cs

```diff
@@ -1593,12 +1588,12 @@ public partial class TextAnalyticsClient
-        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.AI.TextAnalytics.DynamicClassifyDocumentResultCollection>> DynamicClassifyBatchAsync(System.Collections.Generic.IEnumerable<string> documents, System.Collections.Generic.IEnumerable<string> categories, string language = null, Azure.AI.TextAnalytics.DynamicClassifyOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response<Azure.AI.TextAnalytics.ClassificationCategoryCollection> DynamicClassify(string document, System.Collections.Generic.IEnumerable<string> categories, Azure.AI.TextAnalytics.ClassificationType? classificationType = default(Azure.AI.TextAnalytics.ClassificationType?), string language = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.AI.TextAnalytics.ClassificationCategoryCollection>> DynamicClassifyAsync(string document, System.Collections.Generic.IEnumerable<string> categories, Azure.AI.TextAnalytics.ClassificationType? classificationType = default(Azure.AI.TextAnalytics.ClassificationType?), string language = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response<Azure.AI.TextAnalytics.DynamicClassifyDocumentResultCollection> DynamicClassifyBatch(System.Collections.Generic.IEnumerable<Azure.AI.TextAnalytics.TextDocumentInput> documents, System.Collections.Generic.IEnumerable<string> categories, Azure.AI.TextAnalytics.ClassificationType? classificationType = default(Azure.AI.TextAnalytics.ClassificationType?), Azure.AI.TextAnalytics.TextAnalyticsRequestOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response<Azure.AI.TextAnalytics.DynamicClassifyDocumentResultCollection> DynamicClassifyBatch(System.Collections.Generic.IEnumerable<string> documents, System.Collections.Generic.IEnumerable<string> categories, Azure.AI.TextAnalytics.ClassificationType? classificationType = default(Azure.AI.TextAnalytics.ClassificationType?), string language = null, Azure.AI.TextAnalytics.TextAnalyticsRequestOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
DynamicClassifyOptions can extend TextAnalyticsRequestOptions.
In java, we prefer to have a feature-based optional bag DynamicClassifyOptions, because it could add more properties that only apply to  Dynamic Classification but not TextAnalyticsRequestOptions which is the base optional bag that applies to most of TA features.

Reply:
@heaths Should we do this in .NET too? I'm looking at the other methods in the `TextAnalyticsClient` class and I see many of them receive a simple `TextAnalyticsRequestOptions` parameter (e.g. `DetectLanguageBatch`, `ExtractKeyPhrasesBatch`, `RecognizeEntitiesBatch`, etc.). Is it a better idea to create a `DynamicClassifyOptions` class that inherits from `TextAnalyticsRequestOptions` (even if it doesn't add anything to it) like Java does?

Reply:
For more info: https://apiview.dev/Assemblies/Review/bd92063fce2d4a8c933c61ada05c8ce9#com.azure.ai.textanalytics.models.DynamicClassificationOptions

Reply:
WE typically do this in .NET, too, but IIF the derivative class shares _all_ properties of the parent. In TA, there's quite a few cases where that's not true. So, @joseharriaga, if this does share all the parent properties that's fine. You might've been looking at some cases where previous maintainers including myself didn't inherit because not all parent properties were legit. It should follow derivation based on swagger, really.

Reply:
In this case, all the properties in `TextAnalyticsRequestOptions` are valid. In other words, it's not a problem per se, but rather just wondering if we have a preference for using `TextAnalyticsRequestOptions` as the parameter type, or if we prefer to create individual option classes that derive from it for each different feature and use that instead. 

For example, in this case, I would end up with a `DynamicClassifyOptions` class that inherits from `TextAnalyticsRequestOptions` (but does not actually add anything on top of it), and I would change the parameter type of these DynamicClassify- methods to receive a `DynamicClassifyOptions` parameter specifically.